core/src/main/java/com/linecorp/armeria/common/util/LimitedBlockingTaskExecutor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+/**
+ * Provides an executor interface which
+ * is used for recognizing the status of the count of the tasks meets the given limit.
+ *
+ * <p>To throttle using the limit, please refer {@code BlockingTaskLimitingThrottlingStrategy}.
+ */
+public interface LimitedBlockingTaskExecutor extends BlockingTaskExecutor {
```

Comment:
Instead of introducing a new interface, how about providing this feature in `BlockingTaskExecutor`? We could:

- Rename `hitLimit` to `numPendingTasks` and move it up to `BlockingTaskExecutor`.
  - Incorporate what we did in `DefaultLimitBlockingTaskExecutor` into `DefaultBlockingTaskExecutor`.
  - As a result, all `BlockingTaskExecutor` will have proper `numPendingTasks` implementation.

After that, we could update our `ThrottlingStrategy` implementation to accept `IntSupplier` or `int` as its limit and to use `BlockingTaskExecutor.numPendingTasks()` to check if we hit the limit.